src/kinetics/Reaction.cpp

```diff
@@ -38,19 +38,34 @@ Reaction::Reaction(const Composition& reactants_,
     setRate(rate_);
 
     // set flags ensuring correct serialization output
-    bool count = 0;
+    Composition third;
     for (const auto& [name, stoich] : reactants) {
         if (products.count(name)) {
-            count = true;
-            break;
+            third[name] = products.at(name) - stoich;
         }
     }
-    if (count) {
-        if (tbody_ && tbody_->name() != "M") {
+    if (tbody_) {
+        string name = tbody_->name();
+        if (reactants.count("M") || products.count("M")) {
+            throw CanteraError("Reaction::Reaction",
+                "Third body 'M' must not be included in reactant or product lists.");
+        } else if (reactants.count(name)
+                   && reactants.count(name) == products.count(name)) {
+            throw CanteraError("Reaction::Reaction",
+                "Third body '{}' must not have same stoichiometry in reactant and "
+                "product lists.", name);
```

Comment:
This is not checking the stoichiometry of `name` -- only that it exists in both the `reactants` and `products`.

Reply:
This is actually the intent. The logic can still be simplified: `reactants.count(name) && products.count(name)` - I updated the error message to be less confusing: `"'{<name>}' not acting as third body collider must not be included in both reactant and product maps."`